### ISSUE TITLE
MQT fission function calls failing when the load is high

### ISSUE THREAD

**Issue by rakeshbshukla** [fission/fission#1578]
Fission function is deployed using **newdeploy with minscale of 3 and maxscale 40**.
The mqt to function execution is working fine with normal load but it starts failing when there is sudden spike in the incoming load. The scaleup happens eventually but in the mean while lot of messages are dropped as function execution fails with below error
calls failing with 502 giving below error 
```
{"level":"error","ts":1586939151.7698064,"logger":"triggerset.http_trigger_set.metrics-enrichment","caller":"router/functionHandler.go:650","msg":"**error sending request to function","error":"read tcp 10.244.2.127:39092->10.103.105.239:80: read: connection timed out**","function":{"metadata":{"name":"metrics-enrichment","namespace":"fission-function","selfLink":"/apis/fission.io/v1/namespaces/fission-function/functions/metrics-enrichment","uid":"1200447f-90c4-442f-89d7-c4674a2e4cc6","resourceVersion":"11523576","generation":1,"creationTimestamp":"2020-04-15T06:33:52Z","annotations":{"fission-name":"metrics-enrichment","fission-uid":"41bed072-c7ce-46ae-8bd4-77de8211db1c"}},"spec":{"environment":{"namespace":"fission-function","name":"monitoring-core"},"package":{"packageref":{"namespace":"fission-function","name":"metrics-enrichment-py-9ogv","resourceversion":"10748792"}},"secrets":null,"configmaps":[{"namespace":"fission-function","name":"metrics-enrichment-cm"}],"resources":{},"InvokeStrategy":{"ExecutionStrategy":{"ExecutorType":"newdeploy","MinScale":3,"MaxScale":25,"TargetCPUPercent":80,"SpecializationTimeout":0},"StrategyType":"execution"},"functionTimeout":180}},"request_header":{"Accept-Encoding":["gzip"],"Content-Length":["669"],"Content-Type":["application/json"],"Forwarded":["host=router.fission;"],"User-Agent":["Go-http-client/1.1"],"X-Fission-Full-Url":["/fission-function/fission-function/metrics-enrichment"],"X-Fission-Function-Name":["metrics-enrichment"],"X-Fission-Function-Namespace":["fission-function"],"X-Fission-Function-Resourceversion":["11523576"],"X-Fission-Function-Uid":["1200447f-90c4-442f-89d7-c4674a2e4cc6"],"X-Fission-Mqtrigger-Errortopic":[""],"X-Fission-Mqtrigger-Resptopic":[""],"X-Fission-Mqtrigger-Topic":["influx_metrics"],"X-Forwarded-For":["10.244.2.128"],"X-Forwarded-Host":["router.fission"]}} 
```

We have **metrics server** installed and cluster has -**-horizontal-pod-autoscaler-sync-period=5s**

```
**fission version**
[xxxx@xxxxxmanifests]# fission version
client:
  fission/core:
    BuildDate: "2020-02-03T08:40:57Z"
    GitCommit: bda974a72c9093e241c1dae6a7fc1a2d16e28b02
    Version: 1.8.0
server:
  fission/core:
    BuildDate: "2020-02-03T08:40:57Z"
    GitCommit: bda974a72c9093e241c1dae6a7fc1a2d16e28b02
    **Version: 1.8.0**

**kubernetes version:**
[xxx@xxx manifests]# kubectl version
Client Version: version.Info{Major:"1", **Minor:"16", GitVersion:"v1.16.1"**, GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T17:01:15Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", **Minor:"16", GitVersion:"v1.16.1"**, GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T16:51:36Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Baremetal with Red Hat Enterprise Linux Server release 7.7 (Maipo)**

**Comment by vishal-biyani** [comment]
Hey @rakeshbshukla Thanks for raising the issue. I will have someone look into this on priority.

**Comment by vishal-biyani** [comment]
Hey, @rakeshbshukla This is router log, do you also have executor log from the same timeframe? Also, there are two IP addresses mentioned there: 10.103.105.239 and 10.244.2.127 - at that point in time, if you can confirm whose IP addresses are those would be helpful?

**Comment by rakeshbshukla** [comment]
@vishal-biyani  mailed you the detail logs for both router and executor over email along with the requested details. Thanks for the quick help :)

**Comment by vishal-biyani** [comment]
As discussed in the email we modified and deployed https://github.com/fission/fission-kafka-sample and ran some tests. 

We tested in a lab setup that has 4 node GKE cluster setup with Kafka MQT. Each node is n1-standard-1 - so 1CPU & 4GB RAM. We were able to process 250 messages per second (15000) messages per minute. The tests ran for 60 minutes without any issues/errors and no message was dropped. 

Based on logs provided - we suspect this to be a networking resource contention issue in the environment. Specifically the following entries:

```
Failed to delete kube-system/kube-dns:dns endpoint connections, error: error deleting conntrack entries for UDP peer
```


```
"error":"Post http://router.fission/fission-function/fission-function/func-name: dial tcp: i/o timeout"
```

```
dial tcp: lookup executor.fission on 10.96.0.10:53: no such host"
```
